spyder/plugins/projects/tests/test_plugin.py

```diff
@@ -120,5 +127,33 @@ def test_open_project_uses_visible_config(projects, tmpdir, value):
     assert projects.dockwidget.isVisible() == value
 
 
+def test_set_get_project_filenames(projects, tmpdir, mocker):
+    """
+    Test that the currently opened files in the Editor are saved and loaded
+    correctly to and from the project config when the project is closed and
+    then reopened.
+
+    Regression test for Issue #8375
+    """
+    opened_files = ['file1', 'file2', 'file3']
+    mocker.patch.object(
+        projects.main.editor, 'get_open_filenames', return_value=opened_files)
+
+    # We mock os.path.isfile so that we do not have to
+    # actually create the files on the disk.
+    mocker.patch('spyder.plugins.projects.api.os.path.isfile',
+                 return_value=True)
+
+    # Create the project.
+    path = to_text_string(tmpdir.mkdir('project1'))
+    projects.open_project(path=path)
+    assert projects.get_project_filenames() == []
+
+    # Close and reopen the project.
+    projects.close_project()
```

Comment:
Although the original bug only occurred on closing and re-opening projects, would it be too much work to test switching projects as well in case there's a regression or edge case in the future?

Reply:
I agree with this too. It shouldn't be that hard.

Reply:
This is a good idea. I will add that.